GDAXSharp/WebSocket/WebSocket.cs

```diff
@@ -0,0 +1,135 @@
+﻿using GDAXSharp.Network;
+using GDAXSharp.Network.Authentication;
+using GDAXSharp.Shared.Types;
+using GDAXSharp.Shared.Utilities.Clock;
+using GDAXSharp.Shared.Utilities.Extensions;
+using GDAXSharp.WebSocket.Models.Request;
+using GDAXSharp.WebSocket.Models.Response;
+using Newtonsoft.Json;
+using SuperSocket.ClientEngine;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.Http;
+using System.Threading;
+using WebSocket4Net;
+
+namespace GDAXSharp.WebSocket
+{
+    public class WebSocket : AbstractRequest, IWebSocket
+    {
+        private const string ApiUri = "wss://ws-feed.gdax.com";
+
+        private const string SandBoxApiUri = "wss://ws-feed-public.sandbox.gdax.com";
+
+        private CancellationToken CancellationToken { get; set; }
+        private ProductType[] ProductTypes { get; set; }
+        private WebSocket4Net.WebSocket WebSocketFeed { get; set; }
+
+        public WebSocket(IAuthenticator authenticator, IClock clock, CancellationToken cancellationToken, bool sandBox = false) : base(authenticator, clock, sandBox)
```

Comment:
so I am still of the opinion here that the user should not have to pass in the `IClock`. to @chrisw000 's point. yes the `HttpRequestMessageService` follows the same pattern but the user doesn't care about a clock when creating a new `GDAXClient`.

in this case here. when the user wants to new up a new `Websocket` they will have to do something like this 

`var webSocket = new WebSocket(authenticator, new Clock(), CancellationToken.None);`

Reply:
You are right, this needs to be changed.

Reply:
Does the IClock only exist for the tests, to allow the test to dictate a fixed time?

If so, would another solution be to just get rid of IClock altogether, and put the GetTime() method (from clock) onto AbstractRequest - which the test could still change?